### PR TITLE
Improve vendor consumer admin

### PR DESCRIPTION
**Pull request by marc0der**
- Allow patching of credentials and candidates
- Reissues consumer token on update
- Improve consumer revocation

**_Best reviewed commit-by-commit_**

### CODE REVIEW

File: app/repos/ConsumerRepo.scala [hunk 1]
```diff
@@ -12,12 +12,19 @@ import scala.concurrent.Future
...
     extends HasDatabaseConfigProvider[JdbcProfile] {
 
-  def persist(c: Consumer): Future[Int] = {
+  def createOrUpdate(c: Consumer): Future[Int] = {
     db.run((for {
-      id <- sql"INSERT INTO credentials(key, token, owner) VALUES (${c.key}, ${c.token}, ${c.name}) RETURNING id"
+      id <- sql"""
+             INSERT INTO credentials(key, token, owner) 
+             VALUES (${c.key}, ${c.token}, ${c.owner}) 
+             ON CONFLICT(owner) DO UPDATE SET key = ${c.key}, token = ${c.token}
```
Comment (helpermethod) [inline app/repos/ConsumerRepo.scala:20]: Very nice!

File: test/steps/ConsumerSteps.scala [hunk 2]
```diff
@@ -12,9 +12,10 @@ class ConsumerSteps extends ScalaDsl with EN with Matchers with ConsumerMarshall
 
   private val ConsumerTokenPattern = """^[a-f0-9]{64}$""".r
 
-  And("""^the (.*) endpoint receives a POST request:$""") { (endpoint: String, json: String) =>
+  And("""^the (.*) endpoint receives a PATCH request:$""") { (endpoint: String, json: String) =>
     val response = Http(AppHost + endpoint)
       .postData(json.stripMargin)
```
Comment (helpermethod) [inline test/steps/ConsumerSteps.scala:17]: The `.postData`, then setting method to `PATCH` is a litle weird, but the lib doesn't seem to support `patchData`, so I guess this is okay 😄

File: app/repos/ConsumerRepo.scala [hunk 3]
```diff
@@ -28,11 +28,7 @@ class ConsumerRepo @Inject() (val dbConfigProvider: DatabaseConfigProvider)
     } yield id).transactionally)
   }
 
-  def deleteByName(name: String): Future[Int] =
-    db.run((for {
-      result1 <- sqlu"DELETE FROM candidates WHERE name = $name"
-      result2 <- sqlu"DELETE FROM credentials WHERE owner = $name"
-    } yield result1 & result2).transactionally)
+  def deleteByOwner(owner: String): Future[Int] = db.run(sqlu"DELETE FROM credentials WHERE owner = $owner")
```
Comment (helpermethod) [inline app/repos/ConsumerRepo.scala:31]: Why not remove the candidates entries?

Comment (helpermethod) [reply app/repos/ConsumerRepo.scala:31]: Okay, DELETE CASCADE, I see 👍

File: test/steps/ConsumerSteps.scala (hunk 2 above)
Comment (marc0der) [reply test/steps/ConsumerSteps.scala:17]: Yep, it's the only way unfortunately.

**Review by helpermethod — APPROVED** [review]
LGTM!

**Pull request merged**